virtme/guest/virtme-sshd-script

```diff
@@ -17,14 +17,22 @@ rm -f /var/run/nologin
 # of host keys and authorized keys without dubious permission errors
 SSH_DIR=/run/sshd
 
+if grep -q "virtme.cache /run/virtme/cache" /proc/mounts; then
+    # If using args.root != '/', virtme.cache will be mounted.
+    # This way it becomes a central place for ssh files (including id_*.pub),
+    # which defaults to $HOME/.cache/virtme-ng/.ssh in the host.
+    SSH_CACHE=/run/virtme/cache
+fi
+
 # use an ssh location INSIDE of the users home directory to communicate pre-
 # created hostkeys and authorized keys
 SSH_HOME="$(getent passwd "${virtme_ssh_user}" | cut -d: -f6)"
-if [ ! -e "${SSH_HOME}" ]; then
+if [ -e "${SSH_HOME}" ]; then
+    SSH_CACHE="${SSH_HOME}"/.cache/virtme-ng/.ssh
+else
     # Setup an arbitrary ssh location, just to be able to start sshd.
-    SSH_HOME="${SSH_DIR}"
+    SSH_HOME="${SSH_CACHE}"
```

Comment:
could it be possible here not to have `SSH_CACHE` defined?

Would it not be safer to have:

```bash
if [ -e "${SSH_HOME}" ]; then
    SSH_CACHE="${SSH_HOME}"/.cache/virtme-ng/.ssh
elif [ -n "${SSH_CACHE}" ]; then
    # Setup an arbitrary ssh location, just to be able to start sshd.
    SSH_HOME="${SSH_CACHE}"
else
   SSH_HOME="${SSH_DIR}"
   SSH_CACHE="${SSH_HOME}"/.cache/virtme-ng/.ssh
fi
```

(or in a different order to avoid setting `SSH_CACHE` twice the same way)

Reply:
Actually, I'm thinking of skipping this for `args.root != '/'`:
```
SSH_HOME="$(getent passwd "${virtme_ssh_user}" | cut -d: -f6)"
```
Because at this point if the same user from the host exist within the `--root` filesystem, it would use the `/.../root/home/$virtme_ssh_user/.cache/virtme-ng/.ssh` as cache... I will see how this part can be improved/simplified.